draft-packages/guidance-block/KaizenDraft/GuidanceBlock/GuidanceBlock.tsx

```diff
@@ -245,21 +253,20 @@ class GuidanceBlock extends React.Component<
   }
 
   bannerClassName(noMaxWidth): string {
-    return classnames(styles.banner, {
+    const {
+      variant = "default",
+      layout = "default",
```

Comment:
Something feels off about this. Isn't this already set in L87-88?

Reply:
It's a strange one, it is defined with the old school static defaultProps but I get a TS issue if I don't declare it there - It's been a while since I've used class components but maybe the function doesn't know what default props are set in the constructor?

Reply:
![Screen Shot 2022-02-15 at 3 04 43 pm](https://user-images.githubusercontent.com/36558508/153990912-1f785ad7-e5e7-4d8c-9d5d-7d2d99663264.png)

@HeartSquared

Reply:
Gross :( Yeah I'm not too familiar with class based components either.

Could you possibly only destructure at the start of the component and use the destructured values directly within this method? It's strange that we are in a way overriding the values by reinitialising the default value. This would turn into something weird if later on someone only updated one of the two places.

Reply:
hmmmm maybe I can update the function to take props then pass it in when `bannerClassName` gets called, then I can de-structure / hopefully without a ts error